File: ledger-service/src/main/java/zachkingcade/dev/ledger/adapter/in/web/dto/journal/JournalEntryDTOEnrichedResponse.java
package zachkingcade.dev.ledger.adapter.in.web.dto.journal;

import java.time.LocalDate;
import java.util.List;

public record JournalEntryDTOEnrichedResponse(
        Long id,
        LocalDate entryDate,
        String description,
        String notes,
        List<JournalLineDTOEnrichedResponse> journalLines
) {
}

File: user-interface-service/src/app/adapter/ledger-service/dto/Journal-entry/JournalEntryDTOEnrichedResponse.ts
import { JournalLineDTOEnrichedResponse } from "./JournalLineDTOEnrichedResponse";

export interface JournalEntryDTOEnrichedResponse {
  id: number;
  entryDate: string;
  description: string;
  notes: string;
  journalLines: JournalLineDTOEnrichedResponse[];
}
